File: src/main/java/hiiragi283/material/api/container/SlotModuleMachine.java
package hiiragi283.material.api.container;

import hiiragi283.material.tile.TileEntityModuleMachine;
import net.minecraft.item.ItemStack;
import net.minecraftforge.items.IItemHandler;
import net.minecraftforge.items.SlotItemHandler;
import org.jetbrains.annotations.NotNull;

public class SlotModuleMachine extends SlotItemHandler {

    private final TileEntityModuleMachine tile;

    public SlotModuleMachine(TileEntityModuleMachine tile, IItemHandler itemHandler, int index, int xPosition, int yPosition) {
        super(itemHandler, index, xPosition, yPosition);
        this.tile = tile;
    }

    @Override
    public boolean isItemValid(@NotNull ItemStack stack) {
        return tile.getItemSlotCounts() >= getSlotIndex() + 1;
    }

}

File: src/main/kotlin/hiiragi283/material/container/SlotModuleMachine.kt
package hiiragi283.material.container

import hiiragi283.material.api.tile.TileEntityModuleMachine
import net.minecraft.item.ItemStack
import net.minecraftforge.items.IItemHandler
import net.minecraftforge.items.SlotItemHandler

class SlotModuleMachine(
    private val tile: TileEntityModuleMachine,
    itemHandler: IItemHandler,
    index: Int,
    xPosition: Int,
    yPosition: Int
) : SlotItemHandler(itemHandler, index, xPosition, yPosition) {
    override fun isItemValid(stack: ItemStack): Boolean = tile.machineProperty.getItemSlots() >= slotIndex + 1

}